src/libcore/intrinsics.rs

```diff
@@ -869,11 +872,12 @@ extern "rust-intrinsic" {
     /// // the original inner type (`&i32`) to the converted inner type
     /// // (`Option<&i32>`), so read the nomicon pages linked above.
     /// let v_from_raw = unsafe {
+    ///     // Ensure the original vector is not dropped.
+    ///     let v_orig: &mut Vec<_> = &mut *std::mem::ManuallyDrop::new(v_orig);
```

Comment:
This is still not consistently using the same pattern for all three cases. For uniformity, the last example IMO should not exploit the fact that it is indeed the last example.

So, I think there should be a third `v_clone`.

Reply:
Makes sense. Wait for it...